src/Hintman/HLint.hs

```diff
@@ -0,0 +1,91 @@
+{- | This module introduces function to work with @HLint@ tool at modified files
+in PRs.
+-}
+
+module Hintman.HLint
+       ( runHLint
+       ) where
+
+import Language.Haskell.HLint4 (Idea, applyHints, autoSettings, parseModuleEx)
+import Network.HTTP.Client (Response (..), httpLbs)
+import Network.HTTP.Client.TLS (newTlsManager)
+import Network.HTTP.Types (Status (..))
+import System.FilePath (takeExtension)
+import Text.Diff.Parse.Types (FileDelta (..), FileDeltas, FileStatus (..))
+
+import Hintman.Core.PrInfo (Branch (..), Owner (..), PrInfo (..), Repo (..))
+
+import qualified Data.Text as T
+
+
+{- | Run @HLint@ on all modified files.
+
+For this we need:
+
+1. Filter all modified files.
+2. Filter only `*.hs` extension files.
+3. Get the content of all modified files.
+4. Run @Hlint@ tool on the content.
+5. Return the list of all 'Idea's.
+
+-}
+runHLint :: MonadIO m => PrInfo -> m [Idea]
+runHLint prInfo@PrInfo{..} = do
+    let modFiles = filter ((==) ".hs" . takeExtension) $ getModifiedFiles prInfoDelta
+    let getContent = downloadFile . createFileDownloadUrl prInfo
+    fileContent <- traverse (\x -> getContent x >>= \y -> pure (x, y)) modFiles
+    foldMapM getFileHLintSuggestions fileContent
+
+-- | Get all modified files in the PR.
+getModifiedFiles :: FileDeltas -> [FilePath]
+getModifiedFiles = map (toString . fileDeltaDestFile) . filter ((/=) Deleted . fileDeltaStatus)
+
+
+getFileHLintSuggestions :: MonadIO m => (FilePath, Maybe ByteString) -> m [Idea]
+getFileHLintSuggestions (fileName, content) = case content of
+    -- TODO: logging
+    Nothing -> [] <$ putStrLn ("Content is empty: " <> fileName)
+    Just (decodeUtf8 -> c) -> do
+        (flags, classify, hint) <- liftIO autoSettings
+        liftIO $ parseModuleEx flags fileName (Just c) >>= \case
+            Right m -> pure $ applyHints classify hint [m]
+            -- TODO: display error in logging
+            Left _e -> error "Hlint failed with some error"
+
+-- | In-memory file download from GitHub PR sources by given URL.
+downloadFile :: MonadIO m => Text -> m (Maybe ByteString)
+downloadFile url = do
+    man <- newTlsManager
+    let req = fromString $ toString url
+    -- log I $ "Attempting to download file from " <> url <> " ..."
+    response <- liftIO $ httpLbs req man
+    let status = statusCode $ responseStatus response
+    let body = responseBody response
+    -- log D $ "Recieved a status code of " <> show status <> " from " <> url
+    case status of
+        200 -> do
+            -- log I $ "Successfully downloaded file from " <> url
```

Comment:
Thanks for keeping comments for logging! They will be useful after we introduce proper logging